GoogleForCreators/web-stories-wp
# Animations: Background Pan and Zoom Effect

## Context
As part of the upcoming new template creation we wanted to create a ken burns effect which was both the pan and the zoom combined. These were the mockups for it:
![Group 5899 (1)](https://user-images.githubusercontent.com/35983235/110546447-b5c21f00-80eb-11eb-85e0-6e3be1da83b8.png)


## Summary
Adds Pan and zoom background animation effect behind `Experimental Animation Effects` feature flag.

## Relevant Technical Choices
both zoom and pan target the leaf media node so they can animate it within the canvas frame. Because of this I had to pull the keyframes off both the pan and zoom and combine to compose this animation vs just nesting top level animation elements.

Had to add some fancy `transform-origin` calculations to accommodate the user moving the bg image and it scaling from the proper origin on the image (while the pan was going on) to keep the image always filling the canvas.

Found the same bug in our normal bg zoom did a quick fix there too while I was at it.

## To-do
Take out from behind feature flag in a following PR. Wanted to get this in experiments so that UX wasn't a blocker and we could review it with the design team together on staging to further define UX. (This follows the requested UX, but want to get it in front of everyone)

## User-facing changes
NA


## Testing Instructions
Go to experiments -> enable `Experimental Animation Effects` & save -> go to editor
Add or alter bg media to use new `PAN AND ZOOM` animation effect. mess around with it and see that it both Pans... and Zooms 🎉  at the same time 🙀 


### QA






This PR can be tested by following these steps:

1.

### UAT
Follow testing instructions above. Hoping to go over this in a quick meeting/call after it's in staging behind a flag, so will discuss enhancement tickets that need to come out of this after that.






This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6379
